src/module_writer.rs

```diff
@@ -309,14 +318,36 @@ impl WheelWriter {
         }
     }
 
+    /// Returns a DateTime representing the value SOURCE_DATE_EPOCH environment variable
+    /// Note that the earliest timestamp a zip file can represent is 1980-01-01
+    fn mtime(&self) -> Result<DateTime> {
+        let epoch: i64 = env::var("SOURCE_DATE_EPOCH")?.parse()?;
+        let dt = OffsetDateTime::from_unix_timestamp(epoch)?;
+        let min_dt = datetime!(1980-01-01 0:00 UTC);
+        let dt = dt.max(min_dt);
+
+        let dt = DateTime::from_time(dt);
+
+        match dt {
+            Ok(dt) => return Ok(dt),
+            Err(_) => return Err(anyhow!("failed to build zip DateTime")),
+        }
```

Comment:
I think it can be simplified to

```suggestion
        let dt = DateTime::from_time(dt).context("Failed to build zip DateTime")?;
        Ok(dt)
```

Reply:
This does not work due to the following error:
```
error[E0599]: the method `context` exists for enum `std::result::Result<zip::DateTime, ()>`, but its trait bounds were not satisfied
   --> src/module_writer.rs:329:42
    |
329 |         let dt = DateTime::from_time(dt).context("Failed to build zip DateTime")?;
    |                                          ^^^^^^^ method cannot be called on `std::result::Result<zip::DateTime, ()>` due to unsatisfied trait bounds
    |
   ::: /home/zmanji/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/result.rs:504:1
    |
504 | pub enum Result<T, E> {
    | --------------------- doesn't satisfy `_: anyhow::Context<zip::DateTime, ()>`
    |
    = note: the following trait bounds were not satisfied:
            `(): anyhow::context::ext::StdError`
            which is required by `std::result::Result<zip::DateTime, ()>: anyhow::Context<zip::DateTime, ()>`
```

I am not sure how to resolve it. I have updated this to satisfy clippy.